app/models/payment.rb

```diff
@@ -9,6 +9,8 @@ def self.valid_sources
   end
 
   validates :source, presence: true, inclusion: { in: valid_sources }
-  validates :account, :amount, presence: true
+  validates :account, presence: true
+  validates :amount, presence: true, numericality: true
+  validates :processing_fee, numericality: true, allow_nil: false
```

Comment:
I was actually going for slightly different validations: `amount` should be not-null, but also not allow zero, (i.e. `present?`). `processing_fee` is allowed to be zero. I did intentionally allow negative values in the case of a refund or something like that.